modules/shared/src/main/scala/org/tessellation/schema/GlobalSnapshotInfo.scala

```diff
@@ -58,6 +58,81 @@ object GlobalSnapshotStateProof {
   }
 }
 
+@derive(encoder, decoder, eqv, show)
+case class GlobalSnapshotInfoV2(
+  lastStateChannelSnapshotHashes: SortedMap[Address, Hash],
+  lastTxRefs: SortedMap[Address, TransactionReference],
+  balances: SortedMap[Address, Balance],
+  lastCurrencySnapshots: SortedMap[Address, Either[Signed[
+    CurrencySnapshot
+  ], (Signed[CurrencyIncrementalSnapshotV1], CurrencySnapshotInfo)]],
+  lastCurrencySnapshotsProofs: SortedMap[Address, Proof]
+) extends SnapshotInfo[GlobalSnapshotStateProof] {
+  def toGlobalSnapshotInfo: GlobalSnapshotInfo =
+    GlobalSnapshotInfo(
+      lastStateChannelSnapshotHashes,
+      lastTxRefs,
+      balances,
+      lastCurrencySnapshots.map {
+        case (k, v) =>
+          (
+            k,
+            v.map {
+              case (Signed(inc, proofs), info) => (Signed(inc.toCurrencyIncrementalSnapshot, proofs), info)
+            }
+          )
+      },
```

Comment:
What are your thoughts on this ?
```
      gs.lastCurrencySnapshots.view.mapValues(v =>
        v.map {
          case (Signed(inc, proofs), info) =>
            (Signed(CurrencyIncrementalSnapshotV1.fromCurrencyIncrementalSnapshot(inc), proofs), info)
        }
      ).to(gs.lastCurrencySnapshots.sortedMapFactory),
```

Reply:
(I pasted that to the wrong place, .... but similar to code later on.

Reply:
@sdavidp what is `sortedMapFactory`?

Reply:
I see. I've already tried `view.mapValues` but didn't know the trick with sorted map factory. thanks!